src/client/array/dc_array.c

```diff
@@ -728,29 +714,25 @@ dc_array_open(tse_task_t *task)
 	tse_task_t		*open_task = NULL, *fetch_task = NULL;
 	daos_obj_open_t		*open_args;
 	struct md_params	*params;
-	daos_ofeat_t		ofeat;
+	enum daos_otype_t	otype;
 	int			rc;
 
 	if (args->open_with_attr)
 		if (*args->cell_size == 0 || *args->chunk_size == 0)
 			D_GOTO(err_ptask, rc = -DER_INVAL);
 
-	ofeat = daos_obj_id2feat(args->oid);
-	if (!(ofeat & DAOS_OF_DKEY_UINT64)) {
-		D_ERROR("Array Dkeys must be UINT64 Typed (OID feats).\n");
-		D_GOTO(err_ptask, rc = -DER_INVAL);
-	}
-	if (!(ofeat & DAOS_OF_KV_FLAT)) {
-		D_ERROR("Array must be of type Flat KV (OID feats).\n");
+	otype = daos_obj_id2type(args->oid);
+	if (!daos_is_array_type(otype)) {
+		D_ERROR("Array must be of type Array (OID type).\n");
 		D_GOTO(err_ptask, rc = -DER_INVAL);
 	}
-	if (!args->open_with_attr && !(ofeat & DAOS_OF_ARRAY)) {
-		D_ERROR("Array Open must have DAOS_OF_ARRAY (OID feats).\n");
+	if (!args->open_with_attr && otype != DAOS_OT_ARRAY) {
+		D_ERROR("Array Open must have DAOS_OT_ARRAY type (OID type).\n");
 		D_GOTO(err_ptask, rc = -DER_INVAL);
 	}
-	if (args->open_with_attr && (ofeat & DAOS_OF_ARRAY)) {
-		D_ERROR("Array open_with_attr must not have DAOS_OF_ARRAY"
-			" (OID feats).\n");
+	if (args->open_with_attr && otype == DAOS_OT_ARRAY) {
```

Comment:
I think with the type changes we made, we should verify that open_with_attr has either:
ARRAY_TYPE or ARRAY_BYTE